### PR TITLE
feat: replaced the panel implementation with a new one to match our pattern library

### PR DESCRIPTION
**Pull request by thuey**
BREAKING CHANGE: The panel implementation has been replaced, including removing props that are not used in the new implementation

This will be used in https://github.com/dequelabs/walnut/issues/2558

image

The old Panel has been renamed to IssuePanel

### CODE REVIEW

File: packages/react/src/components/Panel/index.tsx [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
 
-const Panel = ({
-  className,
-  title = '',
-  actions,
-  children
-}: {
+interface PanelProps extends HTMLAttributes<HTMLElement> {
+  children: ReactNode;
+  heading:
```
Comment (thuey) [inline packages/react/src/components/Panel/index.tsx:7]: I did this to match what we are doing in the Dialog component in order to encourage good semantics and allow customization: https://github.com/dequelabs/cauldron/blob/develop/packages/react/src/components/Dialog/index.tsx#L19-L24

File: packages/react/src/components/Panel/index.tsx [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
+
   return (
-    <div className={classNames('Panel', className)}>
-      <div className="Panel__Header">
-        {title && <div className="Panel__Header-title">{title}</div>}
-        {actions && <div className="Panel__Header-actions">{actions}</div>}
-      </div>
-      <div className="Panel__Content">{children}</div>
-    </div>
+    <section className={classNames('Panel', className)} {...other}>
```
Comment (thuey) [inline packages/react/src/components/Panel/index.tsx:24]: This seems like it should be a section to me. However, I'm open to being corrected.

File: packages/react/src/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -2,9 +2,9 @@ import { Cauldron } from '../../types';
 import React, { useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import { useDidUpdate } from '../../index';
 import Tab from './Tab';
 import { useId } from 'react-id-generator';
+import useDidUpdate from '../../utils/use-did-update';
```
Comment (thuey) [inline packages/react/src/components/Tabs/Tabs.tsx:7]: We were getting a warning about a cyclic dependency. I made a quick tweak to address it.

**Comment by scurker** [comment]
This does show up in a future design for IGT results:

image

Let's discuss this for a path forward next time we're all at standup.

**Comment by thuey** [comment]
> This does show up in a future design for IGT results:

@scurker Yeah, I posted about that in our slack thread:

> I spoke with @aaronpe about this, and we agree that the usage of that parent component ("Your Products") in the pattern library is more like a container than a card. Also, that the current <Panel> component wasn't necessarily intended to be added to the pattern library, since its only known usage is for IGT results. Since the current <Panel> component is unused (and as Josie referenced in [#413](https://github.com/dequelabs/cauldron/issues/413) we were already thinking of getting rid of it), I'd like to totally replace the <Panel> component with one that fits the design patterns we have. I'll include a comment on https://github.com/dequelabs/axe-extension/issues/1771 noting where the source code for the old <Panel> can be found

I'm happy to discuss this in standup, though 👍

**Comment by thuey** [comment]
In standup, we discussed leaving the old Panel's code in place, but renaming it to `IssuePanel`

**Comment by dequejosie** [comment]
I know we don't need dark theme for the web app, but we might need to consider implementing dark theme as well, or at least make the content visible for dark theme. Do we have a design for dark theme panel? @awpearlm 

Screen Shot 2021-12-17 at 4 56 51 PM

File: packages/styles/panel.css [hunk 4]
```diff
@@ -1,54 +1,22 @@
...
+  --panel-heading-color: var(--gray-90);
 }
 
 .Panel {
+  padding: var(--space-large);
   border: 1px solid var(--panel-border-color);
+  border-radius: 3px;
+  box-shadow: 0px 2px 3px rgba(0, 0, 0, 0.15);
+  background-color: var(--panel-background-color);
+  width: 100%;
```
Comment (dequejosie) [inline packages/styles/panel.css:13]: We might be able to remove this line?

File: packages/react/src/components/Panel/index.tsx (hunk 2 above)
Comment (dequejosie) [reply packages/react/src/components/Panel/index.tsx:24]: I agree with you here

Comment (dylanb) [reply packages/react/src/components/Panel/index.tsx:24]: you should `aria-labelledby` this `<section>` by pointing it at the heading (accessibility best practice)

**Review by dylanb — CHANGES_REQUESTED** [review]
just the aria-labelledby on the section

File: packages/react/src/components/Panel/index.tsx (hunk 2 above)
Comment (thuey) [reply packages/react/src/components/Panel/index.tsx:24]: @dylanb Thanks. I looked elsewhere in our apps where we were using `<section>`, and it didn't seem like we were using `aria-labelledby`, which seemed odd to me. I will go ahead and add it here.

**Comment by thuey** [comment]
> I know we don't need dark theme for the web app, but we might need to consider implementing dark theme as well, or at least make the content visible for dark theme. Do we have a design for dark theme panel? @awpearlm
> 
> Screen Shot 2021-12-17 at 4 56 51 PM

Good catch, @dequejosie! I got this from @awpearlm : https://preview.uxpin.com/4052c8b6f05ce03e28813f28deb3a682950c2ac6#/pages/144239874

File: packages/styles/panel.css (hunk 4 above)
Comment (thuey) [reply packages/styles/panel.css]: Yep, I think you're right.

**Review by dylanb — APPROVED** [review]

**Pull request merged**